src/com/google/enterprise/adaptor/filenet/Permissions.java

```diff
@@ -308,6 +309,43 @@ private void processPermissions() {
       }
     }
 
+    @SuppressWarnings("deprecation")  // For PermissionSource.MARKING
+    /** Builds an ACL from Marking Permissions. */
+    private Acl(int constraintMask) {
+      Iterator<?> iter = perms.iterator();
+      while (iter.hasNext()) {
+        AccessPermission perm = (AccessPermission) iter.next();
+        int mask = perm.get_AccessMask();
+        if ((mask & USE_MARKING) != USE_MARKING) {
+          continue;
+        }
+        if (perm.get_AccessType() == AccessType.ALLOW) {
+          if (perm.get_GranteeType() == SecurityPrincipalType.USER) {
+            allowUsers.put(PermissionSource.MARKING, perm.get_GranteeName());
+          } else {
+            allowGroups.put(PermissionSource.MARKING, perm.get_GranteeName());
+          }
+        } else {
+          if (perm.get_GranteeType() == SecurityPrincipalType.USER) {
+            denyUsers.put(PermissionSource.MARKING, perm.get_GranteeName());
+          } else {
+            denyGroups.put(PermissionSource.MARKING, perm.get_GranteeName());
+          }
+        }
+      }
+      boolean authorizeByConstraints =
+          (VIEW_ACCESS_RIGHTS & ~constraintMask) == VIEW_ACCESS_RIGHTS;
+      if (authorizeByConstraints) {
+        allowGroups.put(PermissionSource.MARKING, AUTHENTICATED_USERS);
```

Comment:
I think a TODO here is good for now, to revisit when we feed the group members, but we need to make AUTHENTICATED_USERS a local group, rather than a global one.